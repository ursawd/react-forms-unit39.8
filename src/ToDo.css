.todo-item {
  display: inline-block;
  margin: 5px;
}
.btn {
  height: 25px;
  border-radius: 25%;
}
