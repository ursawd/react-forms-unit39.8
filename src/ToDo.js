//--ToDo Component
import "./ToDo.css";

const ToDo = ({ id, task, completed, handleRemove }) => {
  const remove = () => handleRemove(id);

  return (
    <div>
      <button type="button" onClick={remove} className="btn">
        ❌
      </button>

      <h3 className="todo-item">{task}</h3>
    </div>
  );
};
export default ToDo;
