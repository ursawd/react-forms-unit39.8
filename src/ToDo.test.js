import React from "react";
import { render, fireEvent, queryByText } from "@testing-library/react";
import ToDo from "./ToDo";

//smoke test
test("renders without crashing", function () {
  render(<ToDo />);
});

//snapshot
test("matches snapshot", function () {
  const { asFragment } = render(<ToDo />);
  expect(asFragment()).toMatchSnapshot();
});
