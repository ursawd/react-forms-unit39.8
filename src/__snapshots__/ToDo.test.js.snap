// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="btn"
      type="button"
    >
      ‚ùå
    </button>
    <h3
      class="todo-item"
    />
  </div>
</DocumentFragment>
`;
